cmdstanpy/cmds.py

```diff
@@ -0,0 +1,184 @@
+import os
+import os.path
+import platform
+import subprocess
+import tempfile
+
+from multiprocessing import cpu_count
+from multiprocessing.pool import ThreadPool
+from typing import Dict
+
+from cmdstanpy import CMDSTAN_PATH, TMPDIR
+from cmdstanpy.lib import Model, StanData, RunSet, SamplerArgs, PosteriorSample
+from cmdstanpy.utils import do_command
+
+
+def compile_model(stan_file:str=None, opt_lvl:int=1,
+                      overwrite:bool=False) -> Model:
+    """Compile the given Stan model file to an executable."""
+    if stan_file is None:
+        raise Exception('must specify argument "stan_file"')
+    if not os.path.exists(stan_file):
+        raise Exception('no such stan_file {}'.format(stan_file))
+    path = os.path.abspath(os.path.dirname(stan_file))
+    program_name = os.path.basename(stan_file)
+    model_name = os.path.splitext(program_name)[0]
+
+    hpp_name = model_name + '.hpp'
+    hpp_file = os.path.join(path, hpp_name)
+    if overwrite or not os.path.exists(hpp_file):
+        print('translating to {}'.format(hpp_file))
+        stanc_path = os.path.join(CMDSTAN_PATH, 'bin', 'stanc')
+        cmd = [stanc_path, '--o={}'.format(hpp_file), stan_file]
+        print('stan to c++: make args {}'.format(cmd))
+        do_command(cmd)
+        if not os.path.exists(hpp_file):
+            raise Exception('syntax error'.format(stan_file))
+
+    exe_file = os.path.join(path, model_name)
+    if platform.system().lower().startswith('win'):
```

Comment:
you can use the `.exe` extension on other platforms too ;)

Reply:
just trying to be consistent w/ cmdstan makefile.